onadata/apps/logger/tasks.py

```diff
@@ -25,3 +27,20 @@ def set_entity_list_perms_async(entity_list_id):
         return
 
     set_project_perms_to_object(entity_list, entity_list.project)
+
+
+@app.task(retry_backoff=3, autoretry_for=(DatabaseError, ConnectionError))
+def apply_project_date_modified_async():
+    """
+    Batch update projects date_modified field periodically
+    """
+    project_ids = cache.get(BATCH_PROJECT_IDS_CACHE, {})
+    if not project_ids:
+        return
+
+    # Update project date_modified field in batches
+    for project_id, timestamp in project_ids.items():
+        Project.objects.filter(pk=project_id).update(date_modified=timestamp)
```

Comment:
It's very rare that a project may not exist, but is possible that it could have been hard deleted. I think we need to handle for the case where the project does not exist

Reply:
@kelvin-muchiri The queryset checks for the existence of the project and updates the `date_modified` field if the project exists, and if it doesn't exist the update method will simply not update any rows

Reply:
@KipSigei Oh, I see